src/main/java/com/minecolonies/coremod/colony/Colony.java

```diff
@@ -562,16 +562,21 @@ public void onServerTick(@NotNull final TickEvent.ServerTickEvent event)
     /**
      * Update Subscribers with Colony, Citizen, and AbstractBuilding Views.
      */
-    public void updateSubscribers()
+    private void updateSubscribers()
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') The Cyclomatic Complexity of this method "updateSubscribers" is 12 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)